### PR TITLE
advertise association with default vpc through properties

### LINKED ISSUE #690: Make it easier to identify resources associated with the default VPC
I encountered a use case where I needed to factory reset (nuke) an AWS account daily but ensure that the AWS default VPC (and its parts) were preserved. At the moment, this is possible by running the tool in dry-run mode against the AWS account in question and then filling out the relevant config.

It would be nicer if any resources (such as route tables, subnets, etc.) associated with that default VPC were flagged through a property that can be filtered within the config. This property could be named something like `DefaultVpcAssoc` and would appear as a property in the following way:

```text
us-east-1 - EC2InternetGateway - igw-e6e53d9c - [DefaultVpcAssoc: "true"] - would remove
```

### PR DESCRIPTION
**Pull request by ga-paul-t**
Make it easier to identify resources that are associated with the default VPC within an AWS account. This enables easier filtering, to preserve those resources. Resources that are of interest:

- EC2InternetGateway
- EC2InternetGatewayAttachment
- EC2RouteTable
- EC2Subnet

Association would be identified through a property called `DefaultVpcAssoc` as follows:

```text
eu-west-1 - EC2InternetGateway - igw-04e54fb4f65ad00be - [DefaultVpcAssoc: "false", tag:Name: "sandbox"] - would remove
eu-west-1 - EC2InternetGateway - igw-cda705ab - [DefaultVpcAssoc: "true"] - would remove
eu-west-1 - EC2RouteTable - rtb-0c54296cacbbc7541 - [DefaultVpcAssoc: "false", tag:Name: "public"] - would remove
eu-west-1 - EC2RouteTable - rtb-0d4dab8b11e4b3f1b - [DefaultVpcAssoc: "false", tag:Name: "private"] - would remove
eu-west-1 - EC2RouteTable - rtb-e021f998 - [DefaultVpcAssoc: "true"] - would remove
eu-west-1 - EC2Subnet - subnet-06b50835fc6073208 - [DefaultForAz: "false", DefaultVpcAssoc: "false", tag:Name: "Private"] - would remove
eu-west-1 - EC2Subnet - subnet-3136c47a - [DefaultForAz: "true", DefaultVpcAssoc: "true"] - filtered by config
eu-west-1 - EC2Subnet - subnet-efe5c9b5 - [DefaultForAz: "true", DefaultVpcAssoc: "true"] - filtered by config
eu-west-1 - EC2Subnet - subnet-0a6a70b0ead5efe9b - [DefaultForAz: "false", DefaultVpcAssoc: "false", tag:Name: "Public"] - would remove
eu-west-1 - EC2Subnet - subnet-634ba81a - [DefaultForAz: "true", DefaultVpcAssoc: "true"] - filtered by config
eu-west-1 - EC2InternetGatewayAttachment - igw-04e54fb4f65ad00be -> vpc-0bb7ed4a1758f6ab6 - [DefaultVpcAssoc: "false", tag:igw:Name: "sandbox", tag:vpc:Creator: "*********", tag:vpc:Name: "sandbox-vpc"] - would remove
eu-west-1 - EC2InternetGatewayAttachment - igw-cda705ab -> vpc-18de6f61 - [DefaultVpcAssoc: "true"] - would remove
```

Fixes #690

### CODE REVIEW

File: resources/ec2-internet-gateway-attachments.go [hunk 1]
```diff
@@ -10,11 +10,12 @@ import (
...
-	vpcId   *string
-	vpcTags []*ec2.Tag
-	igwId   *string
-	igwTags []*ec2.Tag
+	svc         *ec2.EC2
+	vpcId       *string
+	vpcTags     []*ec2.Tag
+	igwId       *string
+	igwTags     []*ec2.Tag
+	defVpcAssoc bool
```
Comment (svenwltr) [inline resources/ec2-internet-gateway-attachments.go:18]: I would simply call this default VPC. Also to follow the naming convention, `VPC` should be spelled uppercase.

```suggestion
	defaultVPC bool
```

File: resources/ec2-internet-gateway-attachments.go [hunk 2]
```diff
@@ -81,6 +83,7 @@ func (e *EC2InternetGatewayAttachment) Properties() types.Properties {
 	for _, tagValue := range e.vpcTags {
 		properties.SetTagWithPrefix("vpc", tagValue.Key, tagValue.Value)
 	}
+	properties.Set("DefaultVpcAssoc", e.defVpcAssoc)
```
Comment (svenwltr) [inline resources/ec2-internet-gateway-attachments.go:86]: I would simply strip the `Assoc` part here. Also to follow convention `VPC` should be spelled uppercase.

```suggestion
	properties.Set("DefaultVPC", e.defaultVPC)
```

**Review by svenwltr — COMMENTED** [review]
Hello. This looks good so far. The only issues are about the naming, because they do not follow the Golang naming conventions. Abbreviations should have consistent case. See [Guidelines](https://github.com/golang/go/wiki/CodeReviewComments#initialisms).

I know that this is not done consistently across this project. This is because the Golang naming conventions clash with the AWS SDK one here and we where not very careful checking this.

**Comment by ga-paul-t** [comment]
Thanks for taking the time to review my change. I will push those changes later on today

**Comment by ga-paul-t** [comment]
@svenwltr I have made those suggested changes

**Pull request merged**